src/Raven.Server/Documents/DatabasesLandlord.cs

```diff
@@ -402,17 +404,76 @@ public void DeleteDatabase(string dbName, DeletionInProgressStatus deletionInPro
             {
                 removeLockAndReturn?.Dispose();
             }
-            NotifyLeaderAboutRemoval(dbName, databaseId);
+
+            var lastDeleteIndex = TryGetLastRaftIndexForActualDelete(_serverStore, dbName);
+            NotifyLeaderAboutRemoval(dbName, databaseId, lastDeleteIndex);
+        }
+
+        public static long? TryGetLastRaftIndexForActualDelete(ServerStore serverStore, string databaseName)
+        {
+            var commandType = nameof(DeleteDatabaseCommand);
+            long? index = null;
+
+            using (serverStore.Engine.ContextPool.AllocateOperationContext(out ClusterOperationContext context))
+            using (context.OpenReadTransaction())
+            {
+                foreach (var entry in serverStore.Engine.LogHistory.GetHistoryLogs(context))
+                {
+                    var type = entry[nameof(RachisLogHistory.LogHistoryColumn.Type)].ToString();
+                    var state = entry[nameof(RachisLogHistory.LogHistoryColumn.State)].ToString();
+                    if (type == commandType && state == "Committed")
+                    {
+                        var guid = entry[nameof(RachisLogHistory.LogHistoryColumn.Guid)].ToString();
+                        if (DeleteDatabaseCommandGuidIsValid(guid, out var databaseNameFromCommand, out var nodes) && 
+                            databaseName == databaseNameFromCommand &&
+                            nodes.Contains(serverStore.NodeTag))
+                        {
+                            index = long.Parse(entry[nameof(RachisLogHistory.LogHistoryColumn.Index)].ToString());
+                        }
+                    }
+                }
+            }
+
+            return index;
+        }
+
+        private static bool DeleteDatabaseCommandGuidIsValid(string guid, out string databaseName, out List<string> nodes)
+        {
+            databaseName = string.Empty;
+            nodes = null;
+
+            int index1 = guid.IndexOf('/');
+            if (index1 < 0)
+                return false;
+
+            try
+            {
+                index1++;
+                var index2 = guid.IndexOf('/', index1 + 1);
+                if (index2 < 0)
+                    return false;
+
+                databaseName = guid.Substring(index1, index2 - index1);
+                index2++;
+                nodes = guid.Substring(index2).Split('_').Where(x => x != string.Empty).ToList();
+            }
+            catch (ArgumentOutOfRangeException)
```

Comment:
the DeleteDatabaseCommand command id should look like that: "a6m8me9843jt/db1/A_B_C"
($"{newGuid}/{databaseName}/{string.Join('_', nodes}")

the code can throw in case of invalid guid,
for example:
in case of invalid guid like this "abcd/"
it will throw here:
https://github.com/ravendb/ravendb/blob/4d7fa5bd3f0d98c5289755a12738044dc1afc95f/src/Raven.Server/Documents/DatabasesLandlord.cs#L452

in case of invalid guid like this "abcd/db1/"
it will throw here:
https://github.com/ravendb/ravendb/blob/4d7fa5bd3f0d98c5289755a12738044dc1afc95f/src/Raven.Server/Documents/DatabasesLandlord.cs#L456C22-L456C22